File: OneToOneCallSample/app/src/test/java/com/stringee/apptoappcallsample/stringee/service/CallServiceRetryPolicyTest.java
package com.stringee.apptoappcallsample.stringee.service;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class CallServiceRetryPolicyTest {
    @Test
    public void retriesOnlyWhileForegroundAndSessionIsStillOwned() {
        assertTrue(CallServiceRetryPolicy.shouldRetry(true, true, 0));
        assertFalse(CallServiceRetryPolicy.shouldRetry(false, true, 0));
        assertFalse(CallServiceRetryPolicy.shouldRetry(true, false, 0));
    }

    @Test
    public void retryCountIsBounded() {
        assertTrue(CallServiceRetryPolicy.shouldRetry(true, true, 2));
        assertFalse(CallServiceRetryPolicy.shouldRetry(true, true, 3));
    }
}

File: Kotlin-OneToOneCallSample/app/src/test/java/com/stringee/kotlin_onetoonecallsample/stringee/service/CallServiceRetryPolicyTest.kt
package com.stringee.kotlin_onetoonecallsample.stringee.service

import com.stringee.kotlin_onetoonecallsample.stringee.service.CallServiceRetryPolicy.shouldRetry
import org.junit.Assert
import org.junit.Test

class CallServiceRetryPolicyTest {
    @Test
    fun retriesOnlyWhileForegroundAndSessionIsStillOwned() {
        Assert.assertTrue(shouldRetry(true, true, 0))
        Assert.assertFalse(shouldRetry(false, true, 0))
        Assert.assertFalse(shouldRetry(true, false, 0))
    }

    @Test
    fun retryCountIsBounded() {
        Assert.assertTrue(shouldRetry(true, true, 2))
        Assert.assertFalse(shouldRetry(true, true, 3))
    }
}
